compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -1050,7 +1097,8 @@ public void visit(BLangCompoundAssignment compoundAssignment) {
     public void visit(BLangInvocation invocation) {
         rewriteInvocationExpr(invocation);
         BLangInvokableNode encInvokable = env.enclInvokable;
-        if (encInvokable == null || !invocation.functionPointerInvocation) {
+        if (encInvokable == null || !invocation.functionPointerInvocation ||
+                !env.enclPkg.packageID.equals(invocation.symbol.pkgID)) {
```

Comment:
We could have had cases like this previously too, right? When the default value is a function call of a function from a different module? Can we add such a test also?